SETUP/lint_php_files.sh

```diff
@@ -11,26 +12,32 @@ else
     exit 1
 fi
 
-echo "Checking all .php and .inc files under $BASE_DIR for linting errors..."
-
-for file in $(find $BASE_DIR -name "*.php" -o -name "*.inc" | grep -v vendor); do
+function lint_file()
+{
+    local file=$1
     echo $file
-    OUTPUT=`php -l $file`
-    echo $OUTPUT | grep -q 'No syntax errors detected'
-    if [ $? -ne 0 ]; then
+    OUTPUT=$(php -l $file)
+    if ! echo $OUTPUT | grep -q 'No syntax errors detected'; then
         echo "ERROR: PHP lint failure in $file"
         echo "$OUTPUT"
         exit 1
     fi
 
     OUTPUT=$(file $file)
-    echo $OUTPUT | grep -q 'ASCII'
-    ASCII=$?
-    echo $OUTPUT | grep -q 'UTF-8'
-    UTF8=$?
-    if [ $ASCII -eq 1 -a $UTF8 -eq 1 ]; then
+    if ! echo $OUTPUT | grep -Eq 'ASCII|UTF-8'; then
         echo "ERROR: Unexpected encoding in $file"
         echo "$OUTPUT"
         exit 1
     fi
+}
+
+echo "Checking all .php and .inc files under $BASE_DIR for linting errors..."
+N=$(nproc)
+for file in $(find $BASE_DIR -name "*.php" -o -name "*.inc" | grep -v "^$BASE_DIR/vendor"); do
```

Comment:
`^$BASE_DIR/vendor` doesn't work unless `$BASE_DIR` is passed in fully qualified, which isn't always the case.

Reply:
Ah. OK. I'll undo this because it's an unnecessary (and wrong) tidyup.